src/librustc_codegen_ssa/back/symbol_export.rs

```diff
@@ -129,9 +129,9 @@ fn reachable_non_generics_provider(
                 //
                 // In general though we won't link right if these
                 // symbols are stripped, and LTO currently strips them.
-                if &*name == "rust_eh_personality" ||
-                   &*name == "rust_eh_register_frames" ||
-                   &*name == "rust_eh_unregister_frames" {
+                if name == "rust_eh_personality" ||
```

Comment:
We can probably replace most string literals with valid literals in them, in the compiler, with `sym::`s.

I'd try this if I was in front of my laptop:
```sh
rg '"[_a-zA-Z][_a-zA-Z0-9]*"' src/lib*
```

Reply:
`rust_eh_personality` already is a symbol. We could add the other two. For all potential static symbols there is a decision about whether the identifier is hot enough or frequently used enough for it to be worthwhile, and that decision is not always clear-cut.

"Intern all the static strings" is a possibility, but it would add a lot more due to functions like this one:
https://github.com/rust-lang/rust/blob/10a52c25cad963986cace7a22c167363afca0d74/src/librustdoc/clean/cfg.rs#L337-L402

Reply:
(I should add: I have already converted all the static identifiers that I though were worth converting. Other people may reasonably think that more should be converted.)